tests/typecheck/contrib/auth/test_decorators.yml

```diff
@@ -35,6 +35,28 @@
         @user_passes_test(lambda u: u.get_username().startswith('super'))
         def view_func(request: HttpRequest) -> HttpResponse: ...
         reveal_type(view_func)  # N: Revealed type is "def (request: django.http.request.HttpRequest) -> django.http.response.HttpResponse"
+-   case: user_passes_test_custom_user_model
+    main: |
+        from typing import Union
+        from django.contrib.auth.decorators import user_passes_test
+        from django.contrib.auth.models import AnonymousUser
+        from django.http import HttpRequest, HttpResponse
+        from users.models import User
+        def check_user(user: Union[User, AnonymousUser]) -> bool: ...
+        @user_passes_test(check_user)
+        def view_func(request: HttpRequest) -> HttpResponse: ...
```

Comment:
You need to assert types here, current case is not enough.

What do you expect to happen?

Reply:
Without the changes in the PR mypy will produce a type error on line 46 (because `check_user` doesn't match `Callable[[AbstratBaseUser|AnonymousUser], bool]`), so the test fails. With this PR the signature of `user_passes_test` is changed so there's no type errors and thus the test passes.